## Question 1
* Install Pipenv
* What's the version of pipenv you installed?
* Use --version to find out


In [60]:
!Pipenv --version

pipenv, version 2023.10.3


## Question 2
* Use Pipenv to install Scikit-Learn version 1.3.1
* What's the first hash for scikit-learn you get in Pipfile.lock?

The first hash is: sha256:0c275a06c5190c5ce00af0acbb61c06374087949f643ef32d355ece12c4db043

## Question 3
client = {"job": "retired", "duration": 445, "poutcome": "success"}

What's the probability that this client will get a credit?

In [24]:
import pickle

In [25]:
model_file = 'model1.bin'
dv_file = 'dv.bin' 

In [26]:
with open(model_file,'rb') as f_in:
        model = pickle.load(f_in)

In [27]:
model

LogisticRegression()

In [28]:
with open(dv_file,'rb') as f_in:
        dv = pickle.load(f_in)

In [29]:
dv

DictVectorizer(sparse=False)

In [30]:
client = {"job": "retired", "duration": 445, "poutcome": "success"}

client

{'job': 'retired', 'duration': 445, 'poutcome': 'success'}

In [31]:
X = dv.transform([client])
y_pred = model.predict_proba(X)[0,1]
get_credit = y_pred >= 0.5
result = {
        'get_credit_probability': round(y_pred,3),
        'get_credit': get_credit
    }
print(result)

{'get_credit_probability': 0.902, 'get_credit': True}


the probability that this client will get a credit is **0.902**

## Question 4
client = {"job": "unknown", "duration": 270, "poutcome": "failure"}

What's the probability that this client will get a credit?

In [18]:
import requests

In [21]:
url = 'http://0.0.0.0:9696/predict'

In [34]:
client = {"job": "unknown", "duration": 270, "poutcome": "failure"}

In [35]:
response = requests.post(url, json=client).json()
round(response['get_credit_probability'],3)

0.14

the probability that this client will get a credit is **0.140**

## Question 5
Download the base image svizor/zoomcamp-model:3.10.12-slim. You can easily make it by using docker pull command.

So what's the size of this base image?

the image size is **147MB**

## Question 6
client = {"job": "retired", "duration": 445, "poutcome": "success"}

What's the probability that this client will get a credit now?

In [41]:
import requests

In [54]:
url = 'http://0.0.0.0:82/predict'

In [55]:
client = {"job": "retired", "duration": 445, "poutcome": "success"}

In [59]:
response = requests.post(url, json=client).json()
round(response['get_credit_probability'],3)

0.727

The probability is 0.727 which the most closest option is **0.730**